Topic-4-JSDP/classes.js

```diff
@@ -0,0 +1,74 @@
+class EventEmitter {
+    constructor() {
+        this.events = {};
+    };
+    on (eventName, callback){
+        if (!this.events[eventName]) {
+            this.events[eventName] = [];
+        }
+        this.events[eventName].push(callback);
+    };
+    emit (eventName) {
+        let event = this.events[eventName];
+        event.forEach(callback => {
+            callback.call();
+        });
+    };
+    off (eventName, callback) {
+        let event = this.events[eventName];
+        if (!event) return this;
```

Comment:
**Style**: It's good to have a blank line in between the var declaration, the logic and the return statement. Following that pattern will make your code to be more readable.

For example, 

```
off(eventName, callback) {
  let event = ...
  let otherVariable = ...

  if (condition) {
    // code goes here  
  }

  return something // only if needed  
}
```

Check that I also use two spaces for each indentation, I think is better. But is my personal opinion, you can choose to use four or two.